docs/data/charts/components/JSAnimationCustomization.js

```diff
@@ -0,0 +1,72 @@
+import { BarChart } from '@mui/x-charts/BarChart';
+import * as React from 'react';
+import { useAnimateBarLabel } from '@mui/x-charts/hooks';
+import Stack from '@mui/material/Stack';
+import Button from '@mui/material/Button';
+
+export default function JSAnimationCustomization() {
+  const [key, animate] = React.useReducer((v) => v + 1, 0);
+
+  return (
+    <Stack>
+      <BarChart
+        key={key}
+        xAxis={[{ scaleType: 'band', data: ['A', 'B', 'C'] }]}
+        series={[
+          {
+            type: 'bar',
+            data: [5, 17, 11],
+          },
+          {
+            type: 'bar',
+            data: [0, 12, 15],
+          },
+          {
+            type: 'bar',
+            data: [1, 3, 9],
+          },
+        ]}
+        width={300}
+        height={400}
+        barLabel="value"
+        slots={{ barLabel: AnimatedBarLabel }}
+      />
+      <Button onClick={() => animate()}>Run Animation</Button>
+    </Stack>
+  );
+}
+
+function AnimatedBarLabel(props) {
+  const {
+    seriesId,
+    dataIndex,
+    color,
+    isFaded,
+    isHighlighted,
+    classes,
+    xOrigin,
+    yOrigin,
+    x,
+    y,
+    width,
+    height,
+    layout,
+    skipAnimation,
+    ...otherProps
+  } = props;
+
+  const animatedProps = useAnimateBarLabel({
+    xOrigin,
+    x,
+    yOrigin,
+    y: y - 4,
+    width,
+    height: 0,
+    layout,
+    skipAnimation,
```

Comment:
This is confusing, could we make an api that receives `offset/x/y`? `x: top, y: center, offset: 4`?

Reply:
I can add an explanation to the docs, should I?

But basically the label will be centered inside the bar. By setting the height to 0, we're basically saying "don't center it". The `useAnimateBarLabel` docs mention that the label will be centered. If a user wants to customize this further, then I guess they can use their own animation library or integrate with the exported `useAnimate` from [this PR](https://github.com/mui/mui-x/pull/17162).